packages/cojson/src/coValueCore/coValueCore.ts

```diff
@@ -528,26 +530,31 @@ export class CoValueCore {
       );
 
       if (result.isOk()) {
-        this.updateCurrentContent();
+        this.processNewTransactions(sessionID, newTransactions);
         this.scheduleNotifyUpdate();
       }
 
       return result;
     });
   }
 
-  private updateCurrentContent() {
```

Comment:
Renamed to `processNewTransactions` because it fits better the responsibility.

In the future we could feed these new transactions directly to the validation pipeline, to skip the iteration on the session log on each update.